src/unix/users.rs

```diff
@@ -80,18 +80,25 @@ pub(crate) unsafe fn get_user_groups(
     group_id: libc::gid_t,
 ) -> Vec<Group> {
     let mut buffer = Vec::with_capacity(2048);
-    let mut groups = Vec::with_capacity(256);
+    const REALLOC_SIZE: usize = 256;
+    // We need to keep track of the number of items in `groups` using this `size` variable because
+    // we can't use `Vec::capacity` for this.
+    //
+    // See https://github.com/GuillaumeGomez/sysinfo/issues/1145 for more information.
+    let mut size = REALLOC_SIZE;
+    let mut groups = Vec::with_capacity(size);
 
     loop {
-        let mut nb_groups = groups.capacity();
+        let mut nb_groups = size;
         if getgrouplist(
             name,
             group_id as _,
             groups.as_mut_ptr(),
             &mut nb_groups as *mut _ as *mut _,
         ) == -1
         {
-            groups.reserve(256);
+            groups.reserve(REALLOC_SIZE);
```

Comment:
`reserve` is not doing anything here because it will only act when `additional > .capacity() - .len()`.
So the key change that is missing here is to modify the length via `groups.set_len(nb_groups as _)` before calling `reserve()` .

I recommend adding some logging before calling `reserve` and setting `REALLOC_SIZE = 1` to make the issue more clear.
```
            println!("before reserve: capacity: {} length: {}", groups.capacity(), groups.len());
            groups.reserve(1);
            println!("after reserve: capacity: {} length: {}", groups.capacity(), groups.len());
```

The current change is causing memory access violation because `nb_groups` gets larger than the capacity of `groups`.